components/micro-gateway-core/src/main/ballerina/gateway/filters/throttle_filter.bal

```diff
@@ -225,53 +220,62 @@ function isApplicationLevelThrottled(AuthenticationContext keyValidationDto) ret
-        isBlockConditionExist(keyvalidationResult.username)){
+    isBlockConditionExist(ipLevelBlockingKey) || isBlockConditionExist(appLevelBlockingKey)) ||
+    isBlockConditionExist(keyvalidationResult.username)) {
         return true;
     } else {
         return false;
     }
 }
 
 function generateThrottleEvent(http:Request req, http:FilterContext context, AuthenticationContext keyValidationDto)
-             returns (RequestStreamDTO) {
-    RequestStreamDTO requestStreamDto = new;
-    string apiVersion = getAPIDetailsFromServiceAnnotation(reflect:getServiceAnnotations
-        (context.serviceRef)).apiVersion;
+    returns (RequestStreamDTO) {
+    RequestStreamDTO requestStreamDto = {};
+    string? apiVersion = getAPIDetailsFromServiceAnnotation(reflect:getServiceAnnotations(context.serviceRef)).apiVersion;
     requestStreamDto.messageID = <string>context.attributes[MESSAGE_ID];
-    requestStreamDto.apiKey = getContext(context) + ":" + apiVersion;
+    requestStreamDto.apiKey = getContext(context);
     requestStreamDto.appKey = keyValidationDto.applicationId + ":" + keyValidationDto.username;
-    requestStreamDto.subscriptionKey = keyValidationDto.applicationId + ":" + getContext(context) + ":" +
-        apiVersion;
+    requestStreamDto.subscriptionKey = keyValidationDto.applicationId + ":" + getContext(context);
     requestStreamDto.appTier = keyValidationDto.applicationTier;
     requestStreamDto.apiTier = keyValidationDto.apiTier;
     requestStreamDto.subscriptionTier = keyValidationDto.tier;
-    requestStreamDto.resourceKey = getContext(context) + "/" + getAPIDetailsFromServiceAnnotation(reflect:
-            getServiceAnnotations(context.serviceRef)).apiVersion;
-    TierConfiguration tier = getResourceLevelTier(reflect:getResourceAnnotations(context.serviceRef,
-            context.resourceName));
-    requestStreamDto.resourceTier = tier.policy;
+    requestStreamDto.resourceKey = getContext(context);
+    TierConfiguration? tier = getResourceLevelTier(reflect:getResourceAnnotations(context.serviceRef,
+    context.resourceName));
+    string? policy = tier.policy;
+    if (policy is string) {
+       requestStreamDto.resourceTier = policy;
+    }
     requestStreamDto.userId = keyValidationDto.username;
     requestStreamDto.apiContext = getContext(context);
-    requestStreamDto.apiVersion = apiVersion;
+    if (apiVersion is string) { 
+        requestStreamDto.apiVersion = apiVersion;
+    }
     requestStreamDto.appTenant = keyValidationDto.subscriberTenantDomain;
     requestStreamDto.apiTenant = getTenantDomain(context);
     requestStreamDto.apiName = getApiName(context);
     requestStreamDto.appId = keyValidationDto.applicationId;
 
+    if (apiVersion is string) {
+        requestStreamDto.apiKey += ":" + apiVersion;
+        requestStreamDto.subscriptionKey +=":" + apiVersion;
+        requestStreamDto.resourceKey += "/" + apiVersion;
+    }
+
```

Comment:
Why are we setting api version to these values

Reply:
Thats how the code was before. I handled the optional string value.